File: cases/taint/go/closure-capture-positive/closure_capture_positive.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: closure-capture-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: closure-capture-sink

func makeHandler() func() {
	captured := dfb_source() // DFB-WITNESS: closure-capture-store
	return func() {
		dfb_sink(captured)
	}
}

func run() {
	handler := makeHandler()
	handler()
}

File: cases/taint/javascript/closure-capture-positive/ClosureCapturePositive.js
function dfb_source() { // DFB-SOURCE: closure-capture-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: closure-capture-sink

function makeHandler() {
  const captured = dfb_source(); // DFB-WITNESS: closure-capture-store
  return () => {
    dfb_sink(captured);
  };
}

function run() {
  const handler = makeHandler();
  handler();
}
